src/dasi/core/AggregatedReadHandle.cc

```diff
@@ -0,0 +1,63 @@
+
+#include "dasi/core/AggregatedReadHandle.h"
+
+#include "dasi/util/Exceptions.h"
+
+#include <iostream>
+#include <ostream>
+
+namespace dasi::core {
+
+AggregatedReadHandle::AggregatedReadHandle(const std::vector<ReadHandle*>& handles) :
+    handles_(handles.begin(), handles.end()),
+    current_(handles_.begin()) {}
+
+AggregatedReadHandle::~AggregatedReadHandle() {
+    if (!handles_.empty()) {
+        std::cerr << "ERROR: " << *this << " has not been closed" << std::endl;
+    }
+}
+
+size_t AggregatedReadHandle::read(void* buf, size_t len) {
+
+    if (current_ == handles_.end()) {
+        return 0;
+    }
+
+    size_t nread = 0;
```

Comment:
I don't think that nread is a good variable name here - as it is ambiguous between the number of handles read, and the number of bytes read.

I might call it bytes_read.